### PR TITLE
Remove lumi lists handling in favor of wmagentFormat LumiList

### LINKED ISSUE #6536: Redo patch #6512 with wmagentFormat LumiList
and fix any other location where we convert the string format to LumiList. see #6533

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #6536
tested in my VM and everything looks ok, though I'd like to run a couple of workflows with the second commit in (tons of pylint changes).

### CODE REVIEW

File: src/python/WMCore/WorkQueue/Policy/Start/Block.py [hunk 1]
```diff
@@ -222,12 +221,12 @@ def getMaskedBlocks(self, task, dbs, datasetPath):
                 lfn = fileLumi['logical_file_name']
                 runNumber = str(fileLumi['run_num'])
                 lumis = fileLumi['lumi_section_num']
-                if lumiMask.get(runNumber, False):
-                    if len(set(lumis) & set(lumiMask[runNumber])) > 0:
-                        maskedBlocks.setdefault(block, {})
-                        maskedBlocks[block].setdefault(lfn, LumiList())
-                        lumiList = LumiList(runsAndLumis = {runNumber: lumis})
-                        maskedBlocks[block][lfn] += (lumiList & taskMask)
+                fileMask = LumiList(runsAndLumis={runNumber: lumis})
```
Comment (ericvaandering) [inline src/python/WMCore/WorkQueue/Policy/Start/Block.py]: Isn't the assumption here that each file only contains one run? That's not guaranteed.

File: src/python/WMCore/WorkQueue/Policy/Start/Block.py [hunk 2]
```diff
@@ -222,12 +221,12 @@ def getMaskedBlocks(self, task, dbs, datasetPath):
...
+                commonMask = taskMask & fileMask
+                if commonMask:
+                    maskedBlocks.setdefault(block, {})
+                    maskedBlocks[block].setdefault(lfn, LumiList())
+                    maskedBlocks[block][lfn] += commonMask
```
Comment (ericvaandering) [inline src/python/WMCore/WorkQueue/Policy/Start/Block.py]: Or is this all in a loop ending here so we build it up in the final data structure?

**Comment by ericvaandering** [comment]
Thanks Alan.

File: src/python/WMCore/WorkQueue/Policy/Start/Block.py (hunk 1 above)
Comment (amaltaro) [inline src/python/WMCore/WorkQueue/Policy/Start/Block.py]: `fileLumis` is a list of dicts, where each dict contains those 3 keys and the run_num is a single int number. 
So `fileLumi` will always have a single `run_num`

File: src/python/WMCore/WorkQueue/Policy/Start/Block.py (hunk 2 above)
Comment (amaltaro) [inline src/python/WMCore/WorkQueue/Policy/Start/Block.py]: I guess I answered it above, let me know otherwise please

**Comment by amaltaro** [comment]
@ticoann please review and merge, all my tests (DMWM + LumiMask tests) succeeded, including the pylint code.

**Pull request merged**